virtio-devices/src/thread_helper.rs

```diff
@@ -44,12 +45,19 @@ where
                     return;
                 }
             }
-            std::panic::catch_unwind(AssertUnwindSafe(f))
-                .or_else(|_| {
+            match std::panic::catch_unwind(AssertUnwindSafe(f)) {
+                Err(_) => {
                     error!("{} thread panicked", thread_name);
-                    thread_exit_evt.write(1)
-                })
-                .ok();
+                    thread_exit_evt.write(1).ok();
+                }
+                Ok(r) => {
+                    r.or_else(|e| {
+                        error!("Error running worker: {:?}", e);
+                        thread_exit_evt.write(1)
+                    })
+                    .ok();
```

Comment:
```suggestion
                     if let Err(e) = r {
                         error!("Error running worker: {:?}", e);
                         thread_exit_evt.write(1).ok();
                     }
```
It reads a bit better IMO.

Reply:
Good point. Updated.